JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -0,0 +1,182 @@
+        key(post.id) {
+            post.id to rememberLazyListState()
+        }
+    }
+
+    BoxWithConstraints {
+        val useListDetail = when (getWindowSize(maxWidth)) {
+            WindowSize.Compact, WindowSize.Medium -> false
+            WindowSize.Expanded -> true
+        }
+
+        // Determine which type of the home screen to display
+        val homeScreenType = if (useListDetail) {
+            HomeScreenType.ListArticle
+        } else {
+            when (uiState) {
+                is HomeUiState.HasPosts -> if (uiState.isArticleOpen) {
+                    HomeScreenType.Article
+                } else {
+                    HomeScreenType.List
+                }
+                is HomeUiState.NoPosts -> HomeScreenType.List
+            }
+        }
+
+        when (homeScreenType) {
+            HomeScreenType.ListArticle -> {
+                HomeListDetailScreen(
+                    uiState = uiState,
+                    onToggleFavorite = onToggleFavorite,
+                    onSelectPost = onSelectPost,
+                    onRefreshPosts = onRefreshPosts,
+                    onErrorDismiss = onErrorDismiss,
+                    onInteractWithList = onInteractWithList,
+                    onInteractWithDetail = onInteractWithDetail,
+                    openDrawer = openDrawer,
+                    homeListLazyListState = homeListLazyListState,
+                    articleDetailLazyListStates = articleDetailLazyListStates,
+                    scaffoldState = scaffoldState
+                )
+            }
+            HomeScreenType.List -> {
+                HomeListScreen(
+                    uiState = uiState,
+                    onToggleFavorite = onToggleFavorite,
+                    onSelectPost = onSelectPost,
+                    onRefreshPosts = onRefreshPosts,
+                    onErrorDismiss = onErrorDismiss,
+                    onInteractWithList = onInteractWithList,
+                    openDrawer = openDrawer,
+                    homeListLazyListState = homeListLazyListState,
+                    scaffoldState = scaffoldState
+                )
+            }
+            HomeScreenType.Article -> {
+                // Guaranteed by above condition
+                check(uiState is HomeUiState.HasPosts)
+
+                ArticleScreen(
+                    post = uiState.selectedPost,
+                    onBack = onInteractWithList,
+                    isFavorite = uiState.favorites.contains(uiState.selectedPost.id),
+                    onToggleFavorite = {
+                        onToggleFavorite(uiState.selectedPost.id)
+                    },
+                    lazyListState = articleDetailLazyListStates.getValue(
+                        uiState.selectedPost.id
+                    ),
+                )
+
+                // If we are just showing the detail, have a back press switch to the list.
+                BackHandler {
+                    onInteractWithList()
```

Comment:
Is this enough to provoke back navigation to the list? If so, can you explain a bit more about how this happens in the comment above? or above this line? thank you

Reply:
Added!